src/gluonts/model/naive_2/_predictor.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+from typing import Iterator, Optional
+
+# Third-party imports
+import numpy as np
+import pandas as pd
+
+# First-party imports
+from gluonts.core.component import validated
+from gluonts.dataset.common import Dataset, DataEntry
+from gluonts.model.forecast import SampleForecast, Forecast
+from gluonts.model.predictor import RepresentablePredictor
+from gluonts.evaluation import get_seasonality
+import statsmodels.api as sm
+
+
+def seasonality_test(past_ts_data: np.array, season_length: int):
+    """
+    Test the time-series for seasonal patterns by performing a 90% auto-correlation test:
+
+    As described here: https://www.m4.unic.ac.cy/wp-content/uploads/2018/03/M4-Competitors-Guide.pdf
+    Code based on: https://github.com/Mcompetitions/M4-methods/blob/master/Benchmarks%20and%20Evaluation.R
+    """
+    critical_z_score = 1.645  # corresponds to 90% confidence interval
+    if len(past_ts_data) < 3 * season_length:
+        is_seasonal = False
```

Comment:
Why not just an early return?

Reply:
The R code was like this, so no particular reason, I can change it of course.